File: AlgoExpert/Arrays/Medium/Longest Peak.java
import java.util.*;

class Program {
  public static int longestPeak(int[] array) {
    // Write your code here.
		int longestPeakLength = 0;
		int i = 1;
		
		while (i < array.length - 1) {
			boolean isPeak = array[i - 1] < array[i] && array[i] > array[i + 1];
			if (!isPeak) {
				i++;
				continue;
			}
			
			int leftIndex = i - 2;
			while (leftIndex >= 0 && array[leftIndex] < array[leftIndex + 1]) {
				leftIndex--;
			}
			
			int rightIndex = i + 2;
			while (rightIndex < array.length && array[rightIndex] < array[rightIndex - 1]) {
				rightIndex++;
			}
			
			int currentPeakLength = rightIndex - leftIndex - 1;
			if (currentPeakLength > longestPeakLength)
				longestPeakLength = currentPeakLength;
			i = rightIndex;
		}
		
    return longestPeakLength;
  }
}

File: Python/AlgoExpert/Arrays/Medium/longestPeak.py
def longestPeak(array):
    # Write your code here.
    result = 0
    for i in range(1, len(array) - 1):
        is_peak = array[i - 1] < array[i] and array[i] > array[i + 1]
        if not is_peak:
            continue
        
        left = i - 1
        right = i + 1
        while left > 0 and array[left - 1] < array[left]:
            left -= 1
        while right < len(array) - 1 and array[right] > array[right + 1]:
            right += 1
            
        peak_length = right - left + 1
        if peak_length > result:
            result = peak_length
       
    return result



